pkg/network/dns/resolveconf.go

```diff
@@ -84,6 +84,17 @@ func GetDomainName(searchDomains []string) string {
 	return selected
 }
 
+//GetLongestServiceDomainName returns the longest service search domain entry
+func GetLongestServiceDomainName(searchDomains []string) string {
+	selected := ""
+	for _, d := range searchDomains {
+		if len(d) > len(selected) && strings.Contains(d, ".svc.") {
```

Comment:
> Please place the `svc` key into a const and ref from where it came.

will add
refereed in PR desc if thats what you mean

> BTW, this is a bit unsafe. Someone may add this as a subdomain by itself or it can be part of the domain regardless.

Right (not as part of the subdomain because the subdomain is the first, and the code here has dot before `svc` as wel, but as part of the domain theoretically)
I think its not a good practice to have it as part of the domain,
we can make it bit more stricter if we will need
Note that even getting the cluster domain doesn't have a clean API method
https://github.com/kubernetes/website/issues/25758  (closed without solution)